File: 594. Longest Harmonious Subsequence/sol.java
import java.util.Arrays;
import java.Math.*;

class Solution {
    public int findLHS(int[] nums) {
        Arrays.sort(nums);
        int prev = nums[0], max = 0, left = 0, right = 0;
        boolean wasChanged = false;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == prev)
                left++;
            else if (nums[i] - prev == 1) {
                wasChanged = true;
                right++;
            } else {
                if (wasChanged) {
                    max = Math.max(max, left + right);
                }
                if (nums[i] - nums[i - 1] == 1) {
                    left = right;
                    right = 1;
                    prev = nums[i - 1];
                    wasChanged = true;
                } else {
                    left = 1;
                    right = 0;
                    prev = nums[i];
                    wasChanged = false;
                }
            }
        }
        if (wasChanged) {
            max = Math.max(max, left + right);
        }
        return max;
    }
}

File: 1695. Maximum Erasure Value/sol.go
package main

func first(nums []int, left, right int) int {
	for i := left; i < right; i++ {
		if nums[i] == nums[right] {
			return i
		}
	}
	return right
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func maximumUniqueSubarray(nums []int) int {
	curr := 0
	_max := 0
	left, right := 0, 0
	for right < len(nums) {
		if f := first(nums, left, right); f == right {
			curr += nums[right]
		} else {
			_max = max(_max, curr)
			for i := left; i < f; i++ {
				curr -= nums[i]
			}
			left = f + 1
		}
		right++
	}
	_max = max(_max, curr)
	return _max
}
